qualtran/bloqs/arithmetic/subtraction.py

```diff
@@ -167,3 +167,93 @@ def _sub_diff_size_regs() -> Subtract:
 _SUB_DOC = BloqDocSpec(
     bloq_cls=Subtract, examples=[_sub_symb, _sub_small, _sub_large, _sub_diff_size_regs]
 )
+
+
+@frozen
+class SubtractFrom(Bloq):
+    """A version of `Subtract` that stores the output in the register being subtracted from.
+
+    Implements $U|a\rangle|b\rangle \rightarrow |a - b\rangle|b\rangle$.
+
+    Args:
+        dtype: Quantum datatype used to represent the integers a, b, and a - b.
+
+    Registers:
+        a: A dtype.bitsize-sized input/output register (register a above).
+        b: A dtype.bitsize-sized input register (register b above).
+    """
```

Comment:
IMO, it would be nice if we have two bloqs for subtract:

1. $|a\rangle |b\rangle \rightarrow |a\rangle |a - b\rangle$
2. $|a\rangle |b\rangle \rightarrow |a\rangle |b - a\rangle$


i.e. in both the cases we preserve the convention that subtraction happens in the second register (`b`) and the first register (`a`) is unmodified. The two cases where you'd want to modify the source instead of target can be reduced by changing the order in which you call the bloq at the call site. 

The approach LGTM otherwise.

Reply:
That seems fine. I changed the signature of this bloq to 2.